### PR TITLE
Added basics for register operands

### PR DESCRIPTION
**Pull request by aka2en**

### CODE REVIEW

File: core/CoreTypes.hpp [hunk 1]
```diff
@@ -2,13 +2,35 @@
...
+
+    static inline const char * const regfile_names[] = {
+        "integer",
+        "float"
+    };
+
+    namespace message_categories {
+        const std::string INFO = "info";
+        // More can be added here, with any identifier...
+    }
```
Comment (klingaard) [inline core/CoreTypes.hpp:27]: Is this used?  I removed it before...

File: core/Inst.hpp [hunk 2]
```diff
@@ -135,13 +137,26 @@ namespace olympia
...
         uint64_t    getRAdr() const        { return target_vaddr_ | 0x8000000; } // faked
         bool        isSpeculative() const  { return is_speculative_; }
 
+        // Rename information
+        RegisterBitMask & getSrcRegisterBitMask(const RegFile rf) {
+            return src_reg_bit_masks_[rf];
+        }
+        RegisterBitMask & getDestRegisterBitMask(const RegFile rf) {
+            return dest_reg_bit_masks_[rf];
+        }
```
Comment (klingaard) [inline core/Inst.hpp:158]: Probably want `const` version of these routines too.

**Review by klingaard — APPROVED** [review]

File: core/CoreTypes.hpp (hunk 1 above)
Comment (aka2en) [reply core/CoreTypes.hpp:27]: Eh, I probably used the outdated file

File: core/Inst.hpp (hunk 2 above)
Comment (aka2en) [reply core/Inst.hpp:158]: Done.

**Pull request merged**